### PR TITLE
Characterisation tests - TRPS0002 and TRPS0011

### PR DESCRIPTION
**Pull request by emadkaramad**
This PR adds characterisation tests for TRPS0002 and TRPS0011 triggers.

### CODE REVIEW

File: characterisation/helpers/generatePhase2Message.ts [hunk 1]
```diff
@@ -46,15 +48,74 @@ type Offence = {
...
+): GeneratePhase2MessageOptions => {
+  if (process.env.USE_BICHARD === "true" && options.messageType === MessageType.PNC_UPDATE_DATASET) {
+    console.error(`
+      ** This test only works in the new Bichard. **
+
+      Old Bichard doesn't populate AnnotatedPNCUpdateDataset object when
+      message type is PNCUpdateDataset and there are no operations and exceptions,
+      resulting in runtime exception when trying to get the correlation ID from AnnotatedPNCUpdateDataset.
+      https://github.com/ministryofjustice/bichard7-next/blob/b82a1533324e5e85984e2cac4f2238524320c226/bichard-backend/src/main/java/uk/gov/ocjr/mtu/br7/errorlistmanager/services/ErrorListServicesImpl.java#L744
+      `)
```
Comment (emadkaramad) [inline characterisation/helpers/generatePhase2Message.ts:72]: Do we want to keep this error log?

**Review by bjpirt — APPROVED** [review]

File: characterisation/triggers/TRPS0011.test.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    await new World({}).db.closeConnection()
+  })
+
+  it("creates a TRPS0011 for AnnotatedHearingOutcome when no operations and exceptions are generated", async () => {
+    const inputMessage = generatePhase2Message({
+      messageType: MessageType.ANNOTATED_HEARING_OUTCOME,
+      hoTemplate: "NoOperationsAndExceptions"
+    })
+
+    const { triggers } = await processPhase2Message(inputMessage, { recordable: false })
```
Comment (chubberlisk) [inline characterisation/triggers/TRPS0011.test.ts:20]: Do we need to set `recordable: false` for all of these? It looks like `recordable` is only used for Phase 1 for mocking the PNC.

File: characterisation/triggers/TRPS0002.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  it("creates a TRPS0002 for AnnotatedHearingOutcome when no operations and exceptions are generated and result code is 3107", async () => {
+    const inputMessage = generatePhase2Message({
+      messageType: MessageType.ANNOTATED_HEARING_OUTCOME,
+      hoTemplate: "NoOperationsAndExceptions",
+      offences: [{ results: [{ cjsResultCode: 3107 }] }]
+    })
+
+    const { triggers } = await processPhase2Message(inputMessage, { recordable: false })
+
+    expect(triggers).toContainEqual({ code })
```
Comment (chubberlisk) [inline characterisation/triggers/TRPS0002.test.ts:23]: A bit of nitpick and preference thing, but I prefer if we hardcoded the trigger code to remove that bit of cognitive load. Personally don't mind the duplication.

```suggestion
    expect(triggers).toContainEqual({ code: TriggerCode.TRPS0002 })
```

**Review by chubberlisk — APPROVED** [review]

**Review by chubberlisk — APPROVED** [review]

**Review by chubberlisk — APPROVED** [review]
:shipit:

**Pull request merged**